### PR TITLE
feat: Curation Overhaul

### PR DESCRIPTION
**Pull request by colin969**
There's a whole lot going on here.

- Library can be chosen. (arcade or theatre, if using Flashpoint Data) Current platform xml logic doesn't make no-library importing possible and needs more directory sanity checks, but won't impact Flashpoint's curators using Infinity or Ultimate.
- Curations are now persistant, located at Flashpoint/Curations/
- Curations now import to a unique folder (extract then move for archive, or copy for folder)
  - Curations may be manually extracted/copied into a folder with any unique name and automatically imported to the Curate tab.
- Curations and Add Apps can now be run from the Curate screen
  - If on Windows, running as admin will use symlinks instead of copying, much faster!
- Progress is now available to any component to use. They must have a unique parent key (usually the name of the page, or name + key of component)
- 
- Components can now keep track of progress data for themselves with ProgressContext
  - 'newProgress' will return a ProgressHandle which can be given out to other functions and components to report back.
  - Check the state with your unique parent key to find any ProgressData's you own that are being updated, use this for rendering.
  - ProgressBar and StatusBar are available for rendering, AutoProgressComponent will chose which one depending on the usePercentDone that the ProgressHandle user has chosen.
  - Util functions are provided in the ProgressDispatch class for ProgressHandle users

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,8 +25,8 @@ Install by running ``sudo apt install libgtk-3-0 libnss3``
 
 ## Development
 Recommended setup for development:
-1. Open a command prompt and run ``npm run build`` and wait for it to finish
-2. In the same command prompt as before, run ``npm run watch`` and let the prompt stay open
+1. Clone the repository with ``git clone --recurse-submodules https://github.com/FlashpointProject/launcher.git launcher``
```
Comment (TBubba) [inline README.md]: This section of the README is about how to build the launcher and how to run that build.
This change should instead be made to the ``Setup`` section above, which is about how to download the project and install the dependencies.

File: src/renderer/Util.ts [hunk 2]
```diff
@@ -122,3 +122,16 @@ export function getFileExtension(filename: string): string {
   if (firstDot === -1) { return ''; }
   return filename.substr(firstDot);
 }
+
+/**
+ * Create a new array and populate it with the properties and values from another array or array like object.
+ * @param arrayLike Array or array like object to copy properties and values from.
+ * @returns New array with the same properties and values as the argument.
+ */
+export function copyArrayLike<T>(arrayLike: { [key: number]: T }): Array<T> {
```
Comment (TBubba) [inline src/renderer/Util.ts]: Is there any reason to keep this function? It is used twice to copy a ``FileList``, but the copies are not used for anything that requires it to be copied (as far as I cal tell).

File: src/renderer/app.tsx [hunk 3]
```diff
@@ -521,6 +528,16 @@ export class App extends React.Component<AppProps, AppState> {
   private updateLanguage = (): void => {
     this.props.langManager.updateContainer();
   }
+
+  private unmountBeforeClose = (): void => {
```
Comment (TBubba) [inline src/renderer/app.tsx]: I'm not sure what the purpose of this is, so please explain it to me.

This function makes refreshing the renderer (by focusing anything in the DevTools sidebar, and pressing F5) exit the launcher instead. I use this method to refresh the renderer heavily in development, to speed up iteration.

File: src/renderer/Util.ts (hunk 2 above)
Comment (colin969) [reply src/renderer/Util.ts]: I was working with the drop functions that already existed, so I moved it to util so I could use it in CurateBox, assuming there was a good reason it was needed to begin with. Can strip it if it's not.

File: README.md (hunk 1 above)
Comment (colin969) [reply README.md]: I genuinely have no idea how I ended up putting it there.

**Comment by colin969** [comment]
Merged into a different PR

**Pull request closed without merging**